src/syft/lib/petlib/__init__.py

```diff
@@ -0,0 +1,40 @@
+# stdlib
+from typing import Any
+from typing import Dict
+from typing import Union
+
+# third party
+import petlib
+
+# syft relative
+from . import ec  # noqa: 401
+from ...ast.globals import Globals
+from .allowlist import allowlist
+
+
+def get_return_type(support_dict: Union[str, Dict[str, str]]) -> str:
+    if isinstance(support_dict, str):
+        return support_dict
+    else:
+        return support_dict["return_type"]
+
+
+def create_petlib_ast(client: Any = None) -> Globals:
+    ast = Globals(client)
+
+    # most methods work in all versions and have a single return type
+    # for the more complicated ones we pass a dict with keys like return_type and
+    # min_version
+    for method, return_type_name_or_dict in allowlist.items():
```

Comment:
You could convert this to the inline version used in supported libraries like `tenseal` since its a bit easier to read and update.

Reply:
Sure, I find allowlist a lot cleaner, but if that is what we follow, I'll make the change